rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -239,23 +247,32 @@ public Integer getValue()
 
     //TODO: Store Node indices too, i.e. split to nodeIndex map and compare indices here instead of strings(nodenames).
 
+    final boolean isValidating = isValidatingCachingBehavior(remotePath);
```

Comment:
Why is this conditional required ?

Reply:
Validation makes calls to BookKeeper methods, which affects metrics values. We need this in order to prevent these metrics from being incremented during validation; otherwise this would skew the metrics counts

Reply:
Got it. I cant think of a better solution than adding if conditions. My worry is that a future change to add metrics will forget to add the if condition. The only minor improvement I can think of is to pass a flag to not change metrics. instead of `isValidatingCachingBehaviour`

Reply:
Feel free to ignore that though. It will not make a massive change to the current approach.